intl/intl.php

```diff
@@ -875,7 +875,7 @@ class NumberFormatter {
      * @param $style
      * @param $pattern [optional]
```

Comment:
Our CI uses these stubs to validate function calls. Because the parameter isn't optional in the actual PHP code (note that a very similar optional `$pattern` parameter right below does have a null default), trying to use the NumberFormatter constructor with only two parameters causes a warning:

```xml
<file name="languages/Language.php">
  <error line="3272" severity="warning" message="Call with 2 arg(s) to \NumberFormatter::__construct() which requires 3 arg(s) defined at vendor/jetbrains/phpstorm-stubs/standard/intl.php:878" source="PhanParamTooFew"/>
</file>
```

Reply:
The pattern parameter is ignored in some cases but if the second parameter is `NumberFormatter::PATTERN_RULEBASED`, it should be set or there would be an exception. That's why I think that providing `NULL` default value will make the signature more confusing. 

And now this parameter is optional for PhpStorm and there are more signatures in stubs where optional parameter is marked as `[optional]` without default value when its not specified. Is it possible to tune your validation to work with `[optional]` annotation? Perhaps it's a proper way to make it work with stubs.